src/advancedHMC_MCMC.jl

```diff
@@ -190,6 +192,66 @@ function physloglikelihood(Tar::LogTargetDensity, θ)
     return physlogprob
 end
 
+# My suggested extra loss function
+function L2loss2(Tar::LogTargetDensity, θ)
+    f = Tar.prob.f
+    dataset = Tar.dataset
+
+    # Timepoints to enforce Physics
+    dataset = Array(reduce(hcat, dataset)')
+    t = dataset[end, :]
+    û = dataset[1:(end - 1), :]
+
+    # parameter estimation chosen or not
+    if Tar.extraparams > 0
+        ode_params = Tar.extraparams == 1 ?
+                     θ[((length(θ) - Tar.extraparams) + 1):length(θ)][1] :
+                     θ[((length(θ) - Tar.extraparams) + 1):length(θ)]
+
+        if length(û[:, 1]) == 1
+            physsol = [f(û[:, i][1],
+                ode_params,
+                t[i])
+                       for i in 1:length(û[1, :])]
+        else
+            physsol = [f(û[:, i],
+                ode_params,
+                t[i])
+                       for i in 1:length(û[1, :])]
+        end
+        #form of NN output matrix output dim x n
+        deri_physsol = reduce(hcat, physsol)
+
+        #   OG deriv(basically gradient matching in case of an ODEFunction)
```

Comment:
I would have expected the loss to be against the solution of the DE instead of the derivative matching here

Reply:
that is already happening, in L2loss

Reply:
The conclusion was that it won't be efficient to simulate PDEs and do this so we'll stick with gradient matching

Reply:
We cannot solve the pde at current params, But we can substitute the data points(observed solutions) into the pde at current params So now the only moving thing params is now bound. We could solve the pde at individual param values during sampling But that is being done already as nn vs dataset output is basically.